baker/algolia/utils/shared.ts

```diff
@@ -61,3 +83,210 @@ export function scaleRecordScores<T extends { score: number }>(
+ * to only include countries in that income group.
+ */
+function filterAvailableEntities(
+    featuredMetric: DbPlainFeaturedMetricWithParentTagName,
+    availableEntities: string[]
+): string[] {
+    if (featuredMetric.incomeGroup === FeaturedMetricIncomeGroup.Default)
+        return availableEntities
+
+    const owidIncomeGroup = getCorrespondingIncomeGroup(
+        featuredMetric.incomeGroup
+    )
+
+    const filteredEntities = availableEntities.filter((entity) => {
+        const country = countriesByName()[entity]
+        if (!country) return false
+        return owidIncomeGroup.members.includes(country.code)
+    })
+
+    return filteredEntities
+}
+
+function generateObjectID(
+    featuredMetric: DbPlainFeaturedMetricWithParentTagName,
+    correspondingRecord: ChartRecord
+): string {
+    return `${correspondingRecord.objectID}-fm-${featuredMetric.incomeGroup}-${slugify(featuredMetric.parentTagName)}`
+}
+
+function getGroupKey(
+    featuredMetric: DbPlainFeaturedMetricWithParentTagName
+): string {
+    return `${featuredMetric.parentTagName}-${featuredMetric.incomeGroup}`
+}
+
+/**
+ * Expands the array with FMs copied from the `default` group
+ * for every income group that currently has zero FMs.
+ *
+ * 1. Copies keep the same `ranking`, `url`, `parentTagName`, etc.
+ * 2. The original `default` FMs are removed from the result –
+ *    they are never pushed to Algolia.
+ */
+function expandDefaultFeaturedMetrics(
+    fms: DbPlainFeaturedMetricWithParentTagName[]
+): ExpandedFeaturedMetric[] {
+    const byTag = groupBy(fms, (fm) => fm.parentTagName)
+    const expanded: ExpandedFeaturedMetric[] = fms.map((fm) => ({
+        ...fm,
+        // Confusing, because there are `default` FMs in this array currently,
+        // but we filter them out once they're expanded.
+        isIncomeGroupSpecificFM: true,
+    }))
+
+    for (const [_, featuredMetricsForTag] of Object.entries(byTag)) {
+        const defaults = featuredMetricsForTag.filter(
+            (fm) => fm.incomeGroup === FeaturedMetricIncomeGroup.Default
+        )
+        // If there are no defaults for this tag, there's nothing to expand
+        if (defaults.length === 0) continue
+
+        const presentGroups = new Set(
+            featuredMetricsForTag
+                .filter(
+                    (fm) => fm.incomeGroup !== FeaturedMetricIncomeGroup.Default
+                )
+                .map((fm) => fm.incomeGroup)
+        )
+
+        for (const group of REAL_FM_INCOME_GROUPS) {
+            if (presentGroups.has(group)) continue
+
+            for (const fm of defaults) {
+                expanded.push({
+                    ...fm,
+                    incomeGroup: group,
+                    isIncomeGroupSpecificFM: false,
+                })
+            }
+        }
+    }
+
+    // Drop the defaults. We only want income-group copies
+    return expanded.filter(
+        (fm) => fm.incomeGroup !== FeaturedMetricIncomeGroup.Default
+    )
```

Comment:
it took me a moment to wrap my head around the "what happens if all income groups are filled in? How is the default case represented?" case, but it eventually made sense looking at the addition of isIncomeGroupSpecificFM to canonic chart and mdim views.

If at least one income group is blank, the the no country case will use the derived set of fm from that group. Otherwise, it will use the canonical view. For the user, there are no visible differences in using either records (thanks to originalAvailableEntities), and only one instance is being returned thanks to Algolia's deduplication.

Makes sense 👍🙂